pkg/service/user.go

```diff
@@ -681,6 +691,32 @@ func (s *userService) deleteUserRoleRelations(ctx context.Context, db bun.IDB, u
 	return ids, nil
 }
 
+func (s *userService) UpdateForceResetFlag(ctx context.Context, user *userv3.User) (*userv3.User, error) {
+	name := user.GetMetadata().GetName()
+	entity, err := dao.GetUserFullByEmail(ctx, s.db, name, &models.KratosIdentities{})
+	if err != nil {
+		return &userv3.User{}, fmt.Errorf("no user found with name '%v'", name)
+	}
+
+	if usr, ok := entity.(*models.KratosIdentities); ok {
+		if usr.IdentityCredential.IdentityCredentialType.Name == "password" {
+			// Don't update details for non local(IDP) users
+			err = s.ap.Update(ctx, usr.ID.String(), map[string]interface{}{
+				"email":      user.GetMetadata().GetName(),
+				"first_name": user.GetSpec().GetFirstName(),
+				"last_name":  user.GetSpec().GetLastName(),
+			}, user.Spec.ForceReset)
+			if err != nil {
+				return &userv3.User{}, err
+			}
+		} else {
+			return &userv3.User{}, fmt.Errorf("unable to update for '%v' due to non IDP user", name)
```

Comment:
do we return a service failure for idp users ? Not sure if this would impact any idp users login

Reply:
Updated. Now this will work for all users irrespective of type of login.